frontend/src/page/Realm.tsx

```diff
@@ -27,8 +56,48 @@ export const Realm: React.FC<Props> = ({ match }) => {
     </React.Fragment>;
 };
 
-const DUMMY_NAMES: { [key: string]: string } = {
-    "lectures": "Lectures",
-    "math": "Department of Mathematics",
-    "algebra": "Algebra I",
+// Looks up each segment of the `/` separated `path` as realm and returns a list
+// of realm ids starting with 0 (root).
+const resolvePath = (path: string): number[] | null => {
```

Comment:
```suggestion
const resolvePath = (path: string): number[] | undefined => {
```

This probably doesn't matter at all, but `undefined` seems to be the more idiomatic type to express optionality.